features/202-session-management/api.md

```diff
@@ -0,0 +1,119 @@
+# Session Management APIs
+
+## SDK API
+```typescript
+// Field value would be empty string if not available
+interface UserAgent {
+    raw: string;
+    name: string; // browser name or app ID
+    version: string; // browser version or app version
+    os: string;
+    osVersion: string;
+    device: string;
+}
+
+// Field value would be empty string if not available
+interface Device {
```

Comment:
If we don't foresee anything more than a name for each device, shall we just use a simple string instead to keep the SDK API simple?

Reply:
If we use a simple string, it would be hard to expand it in future. Some potential information to include:
- GPS location
- device ID
- custom data

Reply:
Hmm, in that case, what is the different between user agent and device? Why not just combine both information into one struct?

Reply:
This struct looks like it fits perfectly?
```
interface Device {
    http_useragent: string;
    name: string;
    version: string
    os: string
    osVersion: string
    deviceModel: string
    deviceName: string
    gps: Coordinates
}
```
}